src/models/player.rs

```diff
@@ -334,15 +338,26 @@ impl<E: Env + 'static> UpdateWithCtx<E> for Player {
                         }
                     } else {
                         Effects::none()
-                    }
+                    };
+                    let push_to_api_effect = if self.api_push_time.is_none()
+                        || E::now() - self.api_push_time.unwrap() >= Duration::seconds(30)
```

Comment:
I'm not sure  `unwrap()` here is justified. What if the DateTime is None?
When do we actually set this DateTime and if we expect for it to always be there, then it should not be `Option`.

Reply:
Also, this should be `match`ed, not `if is_none || .unwrap()`